### PR TITLE
Move some MST constructions to core

### PR DESCRIPTION
**Pull request by ymber**
#### Summary
```SUMMARY: Content "Move some shelter types and a raincatcher from MST to core"```

#### Purpose of change
These ones felt like a safe fit for core and they're quite useful in wilderness games.

#### Describe the solution
Move some shelter types and a raincatcher from MST to core.

### CODE REVIEW

File: data/json/terrain.json [hunk 1]
```diff
@@ -10690,5 +10690,77 @@
...
+  },
+  {
+    "id": "t_leanto",
+    "type": "terrain",
+    "name": "pine lean-to",
+    "symbol": ";",
+    "color": [ "brown" ],
+    "move_cost": 2,
+    "bash": {
+      "str_min": 40,
```
Comment (I-am-Erk) [inline data/json/terrain.json]: Pine lean to should not be more durable than a wood wall

File: data/json/terrain.json [hunk 2]
```diff
@@ -10690,5 +10690,77 @@
...
+      "str_max": 90,
+      "ter_set": "t_dirt",
```
Comment (I-am-Erk) [inline data/json/terrain.json]: This should bash into a pine tree, you shouldn't knock down the whole tree at oncr

File: data/json/terrain.json [hunk 3]
```diff
@@ -10690,5 +10690,77 @@
...
+      ]
+    },
+    "deconstruct": {
+      "ter_set": "t_dirt",
+      "items": [ { "item": "pointy_stick", "count": 4 }, { "item": "string_6", "count": 4 }, { "item": "tarp", "count": 1 } ]
+    },
+    "flags": [ "TRANSPARENT", "FLAMMABLE", "THIN_OBSTACLE", "INDOORS", "MOUNTABLE", "EASY_DECONSTRUCT" ]
+  },
+  {
+    "id": "t_raincatcher",
```
Comment (I-am-Erk) [inline data/json/terrain.json]: This thing should be furniture, so that the underlying terrain type is preserved when you take it down

**Comment by kevingranade** [comment]
We need a more nuanced definition of inside vs outside for a lean-to to be a reasonable addition.

AFAIK throwing one of these up will be exactly the same as a tiny house, which is not appropriate.

**Comment by ymber** [comment]
The lean tos work the same as the t_improvised_shelter that's been in forever. Unless the behavior of the INDOORS flag has changed since I played with that stuff a while back they should behave fine.

**Comment by I-am-Erk** [comment]
That's the understanding I have too. Last I checked, improvised shelters were pretty modest in function, but that was before windbreak changes and heat radiation stuff

**Comment by kevingranade** [comment]
Ok I agree it doesn't look like it's the content at fault but rather the underlying system.
We need to revisit improvised shelters and make them kind of suck.

**Pull request merged**